### PR TITLE
slog: buffer allocation on-demand

### LINKED ISSUE #999: Change standard slog buffer allocation from pre-allocated to on-demand
## Description
Currently, `ObStorageLoggerManager` unconditionally pre-allocates 128 standard slog buffers during startup, each of size 8KB. This results in the consumption of over 1MB of `StorageLoggerM` memory, even when no slog writing is actually performed.

The goal is to change this "standard log buffer pool" to an on-demand allocation model:
- Do not pre-allocate the 128 8KB buffers at startup.
- Retain only the maximum concurrency limit (`MAX_CONCURRENT_ITEM_CNT=128`) and the necessary management structures.
- Allocate a standard buffer only when a slog write operation actually requires one.
- Ensure the following semantics and behaviors remain unchanged:
  - The 8KB standard buffer size.
  - Alignment requirements.
  - The concurrency limit.
  - Buffer reuse and release semantics.
  - The behavior of the `OB_SLOG_REACH_MAX_CONCURRENCY` error.

### PR DESCRIPTION
**Pull request by pfyao1101**
### Task Description

Closes #999.

`ObStorageLoggerManager` previously pre-allocated 128 standard slog buffers during initialization. Each buffer is 8 KB, causing more than 1 MB of `StorageLoggerM` memory to be allocated even when no slog write is performed.

This change allocates standard slog buffers on-demand while preserving the existing buffer size, alignment, concurrency limit, reuse semantics, and error behavior.

### Solution Description

- Initialize only the fixed queue used to manage standard slog buffers during startup.
- Allocate an 8 KB standard slog buffer when the free buffer queue is empty.
- Preserve `LOG_FILE_ALIGN_SIZE` address alignment for newly allocated buffers.
- Protect concurrent arena allocation with `ObSpinLockGuard`.
- Reuse released standard buffers through the existing fixed queue.
- Allocate the slog item before its buffer so that `MAX_CONCURRENT_ITEM_CNT` remains the concurrency limit.
- Roll back the allocated item and buffer if a later allocation or initialization step fails.
- Add unit tests covering lazy allocation, address alignment, buffer reuse, and the maximum concurrency limit.

### Passed Regressions

The following relevant tests passed:

- test_storage_logger_manager: 4/4 passed
- test_storage_log_read_write: 8/8 passed
- test_storage_log_replay: 3/3 passed

### Upgrade Compatibility

This change does not change any public API, slog file format, standard buffer size, alignment requirement, or maximum concurrency limit. No upgrade or migration action is required.

### Other Information

Standard slog buffers are aligned to `LOG_FILE_ALIGN_SIZE` using `upper_align`, preserving the existing buffer alignment requirement.

### Release Note

Reduce slog startup memory usage by allocating standard 8 KB slog buffers on-demand.

### CODE REVIEW

File: src/storage/slog/ob_storage_logger_manager.cpp [hunk 1]
```diff
@@ -209,11 +222,26 @@ int ObStorageLoggerManager::free_item(ObStorageLogItem *log_item)
...
+      if (OB_ISNULL(raw_buffer = static_cast<char *>(allocator_.alloc(alloc_size)))) {
+        ret = OB_ALLOCATE_MEMORY_FAILED;
+        STORAGE_REDO_LOG(WARN, "Fail to allocate log buffer",
+            K(ret), K(alloc_size), LITERAL_K(NORMAL_LOG_ITEM_SIZE));
+      } else {
+        // manually align address
+        log_buffer = reinterpret_cast<char *>(upper_align(
+            reinterpret_cast<int64_t>(raw_buffer),
+            ObLogConstants::LOG_FILE_ALIGN_SIZE));
+        ret = OB_SUCCESS;
```
[the remark this answers is not in the document]
Comment (pfyao1101) [reply src/storage/slog/ob_storage_logger_manager.cpp:243]: I initially tried using `allocator_.alloc_aligned(NORMAL_LOG_ITEM_SIZE, LOG_FILE_ALIGN_SIZE)`, but the alignment unit test showed that the returned address was not always 4 KB aligned.

`ObArenaAllocator::alloc_aligned()` delegates to `PageArena::_alloc_aligned()`. When the current page does not have enough space and the allocator extends or allocates a new page, that path may return `cur_page_->alloc(sz)` without applying the alignment offset again.

The explicit padding and `upper_align` are therefore intentional to guarantee the required 4 KB alignment. The buffers are allocated only on-demand, reused through the free queue, and bounded by the 128 slog items.
